doc/geo-replication.md

```diff
@@ -0,0 +1,200 @@
+# Managing Geo-replication
+
+Geo-replication provides a continuous, asynchronous, and incremental
+replication service from one Gluster Volume to another over Local Area
+Networks(LANs), Wide Area Network (WANs), and across the Internet.
+
+## Replicated Volumes vs Geo-replication
+
+The following table lists the difference between replicated volumes
+and geo-replication:
+
+| Replicated Volumes | Geo-replication |
+| -------------------| --------------- |
+|Mirrors data across clusters | Mirrors data across geographically distributed clusters |
+|Provides high-availability	  | Ensures backing up of data for disaster recovery        |
+|Synchronous replication (each and every file operation is sent across all the bricks)	| Asynchronous replication (checks for the changes in files periodically and syncs them on detecting differences) |
+
+## Prerequisites
+
+### Time Synchronization
+
+On bricks of a geo-replication master volume, all the servers' time
+must be uniform. You are recommended to set up NTP (Network Time
+Protocol) service to keep the bricks sync in time and avoid
+out-of-time sync effect.
+
+For example: In a Replicated volume where brick1 of the master is at
+12.20 hrs and brick 2 of the master is at 12.10 hrs with 10 minutes
+time lag, all the changes in brick2 between this period may go
+unnoticed during synchronization of files with remote volume.
+
+### Master and Remote Volumes
+
+Geo-replication will not create remote volume automatically. To
+establish a Geo-replication session make sure Master and Remote
+volumes are in started state.
+
+**Note**: Glusterd2 will not recognise the volumes created using older
+version of glusterd. Master and Remote volumes should be created using
+glusterd2.
+
+### Firewall, Ports and Remote addresses
+
+- During session create, Geo-replication needs access to remote glusterd2
+  port(Default is `24007`).
+- Once the session is established, Geo-replication workers(one per
+  Master brick) will use `ssh` to sync files.
+- Once the session is established, Master workers will get the remote
+  IP/hostnames from remote volume's info. If the internal IPs are used
+  to create remote volume then Geo-rep workers will not be able to
+  sync the files.
+
+## Creating the session
+
```

Comment:
can we have this as two sections. 1) master and slave on same cluster: 2) master and salve on different cluster.
 will make it easier to understand.

Reply:
ack